luaui/Widgets/gui_gridmenu.lua

```diff
@@ -1160,22 +1163,42 @@ local function gridmenuKeyHandler(_, _, args, _, isRepeat)
 
 			local opts
 
-			if ctrl then
+			if quantity < 0 then
+				quantity = quantity * -1
 				opts = { "right" }
 				Spring.PlaySoundFile(CONFIG.sound_queue_rem, 0.75, "ui")
 			else
 				opts = { "left" }
 				Spring.PlaySoundFile(CONFIG.sound_queue_add, 0.75, "ui")
-			end
 
+				--if quantity is more than 100, more than 20 or more than 5 then use engine logic for better performance (fewer for loops inside queueUnit())
+				if opts ~= { "right" } and not alt and quantity >= 100 then
```

Comment:
`{foo}` always makes a new table so `x ~= {"right"}` is always true. Use numerical constants (`CMD.OPT_RIGHT` etc)